test/extensions/filters/udp/dns_filter/dns_filter_test.cc

```diff
@@ -676,6 +696,40 @@ TEST_F(DnsFilterTest, InvalidShortBufferTest) {
   EXPECT_EQ(DNS_RESPONSE_CODE_FORMAT_ERROR, response_parser_->getQueryResponseCode());
 }
 
+TEST_F(DnsFilterTest, RandomizeFirstAnswerTest) {
+  InSequence s;
+
+  setup(forward_query_off_config);
+  const std::string domain("www.foo16.com");
+
+  const std::string query =
+      Utils::buildQueryForDomain(domain, DNS_RECORD_TYPE_A, DNS_RECORD_CLASS_IN);
+  ASSERT_FALSE(query.empty());
+  sendQueryFromClient("10.0.0.1:1000", query);
+
+  query_ctx_ = response_parser_->createQueryContext(udp_response_);
+  EXPECT_TRUE(query_ctx_->parse_status_);
+  EXPECT_EQ(DNS_RESPONSE_CODE_NO_ERROR, response_parser_->getQueryResponseCode());
+
+  // Although 16 addresses are defined, only 8 are returned
+  EXPECT_EQ(8, query_ctx_->answers_.size());
+
+  // We shuffle the list of addresses when we read the config, and in the case of more than
+  // 8 defined addresses, we randomize the initial starting index. We should not end up with
+  // the first answer being the first defined address, or the answers appearing in the same
+  // order as they are defined. There does exist a non-zero probability that both orders match.
```

Comment:
You control what happens with a mock random, right? Can't we just control the output and then verify against an explicit randomized list?

/wait

Reply:
Yeah, sorry.   I updated the PR.